### ISSUE TITLE
[CSHARP] codegen 2.4 outputting duplicate properties because of inherited model

### ISSUE THREAD

**Issue by 96fauj** [swagger-api/swagger-codegen#8792]
##### Description
I'm using the stable codegen generator for a csharp web api 2.0 project. I'm able to get it to successfully generate the api project. However one of the model classes that is generated has a constructor with duplicate parameters.

##### Swagger-codegen version
2.4.0-SNAPSHOT

##### Swagger declaration file content or url

I don't think including the generated doc will help. I looked at the generated json on my localhosturl/docvs/v1 and there is no duplicate property generated. I also viewed the ui and again no duplicate.

##### Command line used for generation

`java -jar Swagger/swagger-codegen-master/modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate `
-i http://localhost:61995/swagger/docs/v1 `
-l csharp `
-o  BritanniaSwagger `
`

##### Steps to reproduce

We have an inherited class which is applied to multiple viewmodel classes used across the api controllers.
// pseudo, not actual syntax
**Interface**: IAnimal { string: Name, string: Colour }
**Abstract Class**: Mammal : IAnimal { string: Name, string: Colour, bool IsLandAnimal }
**Child Class**: Whale : Mammal { string: Name, int: WeightInKgs, bool IsLandAnimal = true }

The generated c# api proejct generated by the codegen jar generates all the classes that inherit the Abstract classes with 2 "Name" parameters on the constructors.

##### Related issues/PRs

I've trawled through the existing issues, there is some mention of inheritance fixing. One of the posts I came across suggested a similar issue (not this particular issue) was perhaps fixed in a later version. (Might have been an old fix and that issue may have been fixed in an earlier version to 2.4). What I did was pull down the 3.0.1 version and I had other issues with trying to run the command with csharp.

##### Suggest a fix/enhancement

I've tried stepping in to Operation filters and Schema filters and I'm unable to see these duplicated properties on the objects coming through for me to just remove.

**Comment by scavarda** [comment]
Yes, also for me compiler gives many [CS0108](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0108) warnings.
Swagger document is not the problem, it seems to be the csharp client generator

**Comment by dalewking** [comment]
I see the same problem generating for Kotlin, so think it is a more widespread problem

**Comment by dalewking** [comment]
If you want a swagger-spec that demonstrates the problem try the one from [this page](https://api.youneedabudget.com/v1) where [this is the swagger spec](https://api.youneedabudget.com/papi/spec-v1-swagger.json).

**Comment by HugoMario** [comment]
this issue seems to be fixed, i just tried with given files and didn't get the error. So @96fauj  @dalewking  @scavarda . Can you guys please try with this https://github.com/swagger-api/swagger-codegen/releases/tag/v2.4.1 and let me know if there is something wrong?

**Comment by scavarda** [comment]
It seems I'm still getting the same warnings with 2.4.1 maven package.
I also tried the swagger file of @dalewking, but it is not generating derived classes, only base classes.

**Comment by dnltsk** [comment]
Just stumbled over the same issue in the Kotlin template of swagger codegen version 2.4.5.

**Comment by dalewking** [comment]
Once again, the problem is not in the code generator, it was actually in the parser. The current version of openapi no longer has this problem.

**Comment by dnltsk** [comment]
@dalewking "Once again"? Cannot find the first mention. In my build.gradle I have the dependencies
```
swaggerCodegen("io.swagger:swagger-codegen-cli:2.4.5")
compile("io.swagger:swagger-core:1.5.22")
```
How can I change version of the openapi parser?

**Comment by dalewking** [comment]
> "Once again"? Cannot find the first mention

The mention was in the [issue that I linked to above](https://github.com/swagger-api/swagger-core/issues/3049).

The swagger-* tools and the openapi-* tools are too competing set of tools. The swagger tools are from SmartBear. The openApi tools are community-supported open source forks. Originally the OpenAPI tools supported OpenAPI specs (aka Swagger 3.0) while the swagger tools only supported up to Swagger 2. That has since changed and both now support both versions.

There may be a version of swagger-core that fixes this bug, but I know it is fixed in the OpenAPI toolset.

To migrate from the Swagger tools to OpenAPI tools, see [this document](https://github.com/OpenAPITools/openapi-generator/blob/master/docs/migration-from-swagger-codegen.md).

**Comment by dalewking** [comment]
> swaggerCodegen("io.swagger:swagger-codegen-cli:2.4.5")
> compile("io.swagger:swagger-core:1.5.22")

Those are not the latest versions of the swagger toolset. You might see if switching to versions 3.0.0 and 2.0.8, respectively, behave any better. Note that the group-id changes for swagger-codegen between 2.* and 3.* and there may be other changes as well. Or switch to [openapi-generator](https://github.com/OpenAPITools/openapi-generator).

**Comment by HugoMario** [comment]
this has been fixed from codegen version 3.0.25, so closing now.

**Comment by smargoli2** [comment]
I am still having this issue with swagger cli 2.4.28. Trying to generate csharp code for Amazon SP-APIs.
